libs/http/include/http/connection.hpp

```diff
@@ -222,9 +216,17 @@ class HTTPConnection : public AbstractHTTPConnection,
     asio::async_write(socket_, *buffer_ptr, cb);
   }
 
+  void CloseConnnection() override
+  {
+    std::error_code dummy;
```

Comment:
Make sure to cleanly close the connection when the connection manager is finished with it